### ISSUE TITLE
flutter/flutter Cirrus should run flutter/plugins tests

### ISSUE THREAD

**Issue by collinjackson** [flutter/flutter#30446]
Opening this issue to track progress on this desired infra feature that will help identify Flutter changes that break plugins.

**Comment by mklim** [comment]
Talked about this a little with @Hixie offline. There's some blockers to getting our whole suite up and running but an immediate low hanging fruit would be to just add our analyzer warnings and potentially unit tests to flutter/tests. It won't prevent plugins from being truly broken by framework changes but it would be trivial to add and would prevent a lot of surprising breakages based on things like a new const constructor.

**Comment by stuartmorgan** [comment]
@Hixie Can you summarize the blockers here? This would be an alternative to going to a roller for Flutter in flutter/plugins.

**Comment by Hixie** [comment]
I don't think there's any blockers at this point, we just need to do it. We should add a shard to flutter/flutter that checks out the right version of flutter/plugins (I'd use [find_commit.dart](https://github.com/flutter/flutter/blob/master/dev/tools/bin/find_commit.dart), but you can use a pin if you prefer, so long as we have an autoroller — let's not add yet more manual rolls) and runs the tests.

**Comment by Hixie** [comment]
(I'd probably do this on LUCI, not Cirrus.)

**Comment by stuartmorgan** [comment]
We already have the autoroller with a pinned version for the existing analyze-flutter/plugins-in-flutter/flutter test, so we could bootstrap on that.

@godofredoc Anything you can think of that would prevent us from adding this now?

**Comment by stuartmorgan** [comment]
I'm going to close this in favor of https://github.com/flutter/flutter/issues/93811 for solving the remaining out-of-band failures. We're already running analyzer tests in flutter/flutter, and I think doing more than that is probably diminishing returns; running all tests would be problematic for various reasons (duplication of CI resources, maintenance, exacerbation of flake in flutter/flutter, etc.).
